--- a/include/rsl/parameter_validators.hpp
+++ b/include/rsl/parameter_validators.hpp
@@ -30,7 +30,7 @@
 [[nodiscard]] auto join_stringified(std::vector<T> const& values) -> std::string {
     std::vector<std::string> tokens;
     tokens.reserve(values.size());
-    for (auto const& value : values) tokens.push_back(stringify(value));
+    for (auto const& value : values) tokens.push_back(stringify(static_cast<T>(value)));
     return fmt::format("{}", fmt::join(tokens, ", "));
 }
 
@@ -92,9 +92,9 @@
     -> tl::expected<void, std::string> {
     auto const& values = parameter.get_value<std::vector<T>>();
     for (auto const& value : values)
-        if (!contains(valid_values, value))
+        if (!contains(valid_values, static_cast<T>(value)))
             return tl::unexpected(
-                fmt::format("Entry '{}' in parameter '{}' is not in the set '{{{}}}'", value,
+                fmt::format("Entry '{}' in parameter '{}' is not in the set '{{{}}}'", static_cast<T>(value),
                             parameter.get_name(), fmt::join(valid_values, ", ")));
     return {};
 }
@@ -166,7 +166,7 @@
 [[nodiscard]] auto element_bounds(rclcpp::Parameter const& parameter, T const& lower,
                                   T const& upper) -> tl::expected<void, std::string> {
     auto const& param_value = parameter.get_value<std::vector<T>>();
-    for (auto val : param_value)
+    for (T val : param_value)
         if (val < lower || val > upper)
             return tl::unexpected(
                 fmt::format("Value '{}' in parameter '{}' must be within bounds '[{}, {}]'",
@@ -185,7 +185,7 @@
 [[nodiscard]] auto lower_element_bounds(rclcpp::Parameter const& parameter,
                                         T const& lower) -> tl::expected<void, std::string> {
     auto const& param_value = parameter.get_value<std::vector<T>>();
-    for (auto val : param_value)
+    for (T val : param_value)
         if (val < lower)
             return tl::unexpected(fmt::format(
                 "Value '{}' in parameter '{}' must be above lower bound of '{}'",
@@ -203,7 +203,7 @@
 [[nodiscard]] auto upper_element_bounds(rclcpp::Parameter const& parameter,
                                         T const& upper) -> tl::expected<void, std::string> {
     auto const& param_value = parameter.get_value<std::vector<T>>();
-    for (auto val : param_value)
+    for (T val : param_value)
         if (val > upper)
             return tl::unexpected(fmt::format(
                 "Value '{}' in parameter '{}' must be below upper bound of '{}'",
