--- a/include/rsl/algorithm.hpp
+++ b/include/rsl/algorithm.hpp
@@ -15,9 +15,9 @@
  * rsl::contains(values, -1); // false
  * @endcode
  */
-template <typename Collection>
+template <typename Collection, typename T>
 [[nodiscard]] auto contains(Collection const& collection,
-                            typename Collection::const_reference value) {
+                            T const& value) {
     return std::find(collection.cbegin(), collection.cend(), value) != collection.cend();
 }
 
