--- a/test/suicide.cpp
+++ b/test/suicide.cpp
@@ -64,6 +64,8 @@
 	abort_abort_I_repeat_abort_abort();
 }
 
+// aarch64, arm, mips, PowerPC and RISC-V do not trap Division by zero
+#if !defined(__aarch64__) && !defined(__arm__) && !defined(__mips__) && !defined (__powerpc__) && !defined (__riscv)
 volatile int zero = 0;
 
 int divide_by_zero()
@@ -78,7 +80,10 @@
 	int v = divide_by_zero();
 	std::cout << "v=" << v << std::endl;
 }
+#endif
 
+// Darwin does not allow RLIMIT_STACK to be reduced
+#ifndef __APPLE__
 int bye_bye_stack(int i) {
 	return bye_bye_stack(i + 1) + bye_bye_stack(i * 2);
 }
@@ -91,3 +96,4 @@
 	int r = bye_bye_stack(42);
 	std::cout << "r=" << r << std::endl;
 }
+#endif
