--- a/test/_test_main.cpp
+++ b/test/_test_main.cpp
@@ -24,7 +24,6 @@
 #include "test.hpp"
 #include <cstdio>
 #include <cstdlib>
-#include <error.h>
 #include <unistd.h>
 #include <sys/wait.h>
 
@@ -40,7 +39,8 @@
 		exit(static_cast<int>(test.run()));
 	}
 	if (child_pid == -1) {
-		error(EXIT_FAILURE, 0, "unable to fork");
+		fprintf(stderr, "unable to fork\n");
+		exit(EXIT_FAILURE);
 	}
 
 	int child_status = 0;
