--- test/rosbag2_cpp/test_sequential_reader.cpp
+++ test/rosbag2_cpp/test_sequential_reader.cpp
@@ -253,6 +253,9 @@
   const auto bag_path = fs::path(temporary_dir_path_) / "bag_with_no_msgs";
   const std::string topic_name = "topic_with_0_messages";
   const auto storage_id = GetParam();
+  if (storage_id == "mcap") {
+    GTEST_SKIP() << "MCAP does not support topics with zero messages without writing messages.";
+  }
   {
     rosbag2_storage::TopicMetadata topic_metadata;
     topic_metadata.name = topic_name;
