--- a/resource/msg__type_support.cpp.em
+++ b/resource/msg__type_support.cpp.em
@@ -211,7 +211,8 @@
 }  // namespace @(ns)
 @[  end for]@
 
-namespace rosidl_typesupport_protobuf
+/*
+namespace rosidl_typesupport_protobuf_cpp
 {
 
 template<>
@@ -221,8 +222,8 @@
   return &@(namespaced_handle_name);
 }
 
-}  // namespace rosidl_typesupport_protobuf
-
+}  // namespace rosidl_typesupport_protobuf_cpp
+*/
 #ifdef __cplusplus
 extern "C"
 {
