--- test/rosbag2_tests/test_rosbag2_info_end_to_end.cpp
+++ test/rosbag2_tests/test_rosbag2_info_end_to_end.cpp
@@ -32,8 +32,12 @@
 public:
   InfoEndToEndTestFixture()
   {
-    // _SRC_RESOURCES_DIR_PATH defined in CMakeLists.txt
-    bags_path_ = fs::absolute(fs::path(_SRC_RESOURCES_DIR_PATH) / GetParam()).generic_string();
+    const char* test_srcdir = getenv("TEST_SRCDIR");
+    if (test_srcdir != nullptr) {
+      bags_path_ = (fs::path(test_srcdir) / "rosbag2_tests+/resources" / GetParam()).generic_string();
+    } else {
+      bags_path_ = fs::absolute(fs::path(_SRC_RESOURCES_DIR_PATH) / GetParam()).generic_string();
+    }
   }
 
   std::string bags_path_;
