diff --git a/rcl/test/rcl/test_service.cpp b/rcl/test/rcl/test_service.cpp
index 0d637c7..b44bc04 100644
--- a/rcl/test/rcl/test_service.cpp
+++ b/rcl/test/rcl/test_service.cpp
@@ -462,7 +462,8 @@ MOCKING_UTILS_BOOL_OPERATOR_RETURNS_FALSE(rcutils_allocator_t, !=)
 
 /* Test failed service initialization using mocks
  */
-TEST_F(TestServiceFixture, test_fail_ini_mocked) {
+// TODO(asymingt) enable once mocking works in Bazel.
+TEST_F(TestServiceFixture, DISABLED_test_fail_ini_mocked) {
   const rosidl_service_type_support_t * ts = ROSIDL_GET_SRV_TYPE_SUPPORT(
     test_msgs, srv, BasicTypes);
   constexpr char topic[] = "topic";
@@ -526,7 +527,8 @@ TEST_F(TestServiceFixture, test_fail_ini_mocked) {
 
 /* Test failed service finalization using mocks
  */
-TEST_F(TestServiceFixture, test_fail_fini_mocked) {
+// TODO(asymingt) enable once mocking works in Bazel.
+TEST_F(TestServiceFixture, DISABLED_test_fail_fini_mocked) {
   const rosidl_service_type_support_t * ts = ROSIDL_GET_SRV_TYPE_SUPPORT(
     test_msgs, srv, BasicTypes);
   constexpr char topic[] = "primitives";
@@ -550,7 +552,8 @@ TEST_F(TestServiceFixture, test_fail_fini_mocked) {
 
 /* Test failed service take_request_with_info using mocks and nullptrs
  */
-TEST_F(TestServiceFixture, test_fail_take_request_with_info) {
+// TODO(asymingt) enable once mocking works in Bazel.
+TEST_F(TestServiceFixture, DISABLED_test_fail_take_request_with_info) {
   const rosidl_service_type_support_t * ts = ROSIDL_GET_SRV_TYPE_SUPPORT(
     test_msgs, srv, BasicTypes);
   constexpr char topic[] = "primitives";
@@ -618,7 +621,8 @@ TEST_F(TestServiceFixture, test_fail_take_request_with_info) {
 
 /* Test failed service send_response using mocks and nullptrs
  */
-TEST_F(TestServiceFixture, test_fail_send_response) {
+// TODO(asymingt) enable once mocking works in Bazel.
+TEST_F(TestServiceFixture, DISABLED_test_fail_send_response) {
   const rosidl_service_type_support_t * ts = ROSIDL_GET_SRV_TYPE_SUPPORT(
     test_msgs, srv, BasicTypes);
   constexpr char topic[] = "primitives";