diff --git a/rcl/test/rcl/test_context.cpp b/rcl/test/rcl/test_context.cpp
index 55aa485..480c918 100644
--- a/rcl/test/rcl/test_context.cpp
+++ b/rcl/test/rcl/test_context.cpp
@@ -142,7 +142,8 @@ TEST(TestContext, nominal) {
   EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
 }
 
-TEST(TestContext, bad_fini) {
+// TODO(asymingt) enable once mocking works in Bazel.
+TEST(TestContext, DISABLED_bad_fini) {
   EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, rcl_context_fini(nullptr));
   rcl_reset_error();
 