diff --git a/rosidl_generator_py/resource/_action.py.em b/rosidl_generator_py/resource/_action.py.em
index 77fbf87..9ba1251 100644
--- a/rosidl_generator_py/resource/_action.py.em
+++ b/rosidl_generator_py/resource/_action.py.em
@@ -41,7 +41,7 @@ class Metaclass_@(action.namespaced_type.name)(type):
     def __import_type_support__(cls):
         try:
             from rosidl_generator_py import import_type_support
-            module = import_type_support('@(package_name)')
+            module = import_type_support('@(package_name)__action_@(action_name)')
         except ImportError:
             import logging
             import traceback