--- a/ament_index_python/search_paths.py
+++ b/ament_index_python/search_paths.py
@@ -13,6 +13,7 @@
 # limitations under the License.
 
 import os
+from pathlib import Path
 from typing import List
 
 from .constants import AMENT_PREFIX_PATH_ENV_VAR
@@ -27,6 +28,14 @@
     """
     ament_prefix_path = os.environ.get(AMENT_PREFIX_PATH_ENV_VAR)
     if not ament_prefix_path:
+        runfiles_dir = os.environ.get("RUNFILES_DIR")
+        test_workspace = os.environ.get("TEST_WORKSPACE")
+        if runfiles_dir and test_workspace:
+            return [os.path.join(runfiles_dir, test_workspace)]
+        p1 = Path.cwd()
+        p2 = Path("../MANIFEST")
+        if (p1 / p2).is_file():
+            return [str(p1)]
         raise EnvironmentError(
             "Environment variable '{}' is not set or empty".format(AMENT_PREFIX_PATH_ENV_VAR))
 
