--- BUILD.bazel
+++ BUILD.bazel
@@ -23,15 +23,17 @@
 cc_library(
     name = "hdrs",
     hdrs = glob(["include/ogg/*.h"]) + ["include/ogg/config_types.h"],
     includes = ["include"],
+    visibility = ["//visibility:public"],
 )
 
 cc_library(
     name = "libogg",
     srcs = [
         "src/bitwise.c",
         "src/framing.c",
     ],
     textual_hdrs = ["src/crctable.h"],
     deps = [":hdrs"],
+    visibility = ["//visibility:public"],
 )
