--- a/src/os_ros.cpp
+++ b/src/os_ros.cpp
@@ -256,4 +256,29 @@
     return telemetry;
 }
 
+namespace impl {
+
+
+template <typename pixel_type>
+ouster::sdk::core::img_t<pixel_type> load_mask(const std::string& mask_path,
+                                    size_t, size_t) {
+    if (mask_path.empty()) {
+        return ouster::sdk::core::img_t<pixel_type>();
+    }
+    throw std::runtime_error("Loading mask files from disk is not supported");
+}
+
+template ouster::sdk::core::img_t<uint8_t> load_mask<uint8_t>(const std::string&, size_t, size_t);
+template ouster::sdk::core::img_t<uint16_t> load_mask<uint16_t>(const std::string&, size_t, size_t);
+template ouster::sdk::core::img_t<uint32_t> load_mask<uint32_t>(const std::string&, size_t, size_t);
+template ouster::sdk::core::img_t<uint64_t> load_mask<uint64_t>(const std::string&, size_t, size_t);
+template ouster::sdk::core::img_t<int8_t> load_mask<int8_t>(const std::string&, size_t, size_t);
+template ouster::sdk::core::img_t<int16_t> load_mask<int16_t>(const std::string&, size_t, size_t);
+template ouster::sdk::core::img_t<int32_t> load_mask<int32_t>(const std::string&, size_t, size_t);
+template ouster::sdk::core::img_t<int64_t> load_mask<int64_t>(const std::string&, size_t, size_t);
+template ouster::sdk::core::img_t<float> load_mask<float>(const std::string&, size_t, size_t);
+template ouster::sdk::core::img_t<double> load_mask<double>(const std::string&, size_t, size_t);
+
+} // namespace impl
+
 }  // namespace ouster_ros
