diff --git a/Cargo.toml b/Cargo.toml
index b7a72147..d3af5e7d 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -31,7 +31,7 @@ build = "build.rs"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [features]
-shared-memory = ["zenoh/shared-memory"]
+shared-memory = ["zenoh/shared-memory", "zenoh-protocol/shared-memory"]
 unstable = ["zenoh/unstable", "zenoh-ext/unstable"]
 auth_pubkey = ["zenoh/auth_pubkey"]
 auth_usrpwd = ["zenoh/auth_usrpwd"]
@@ -52,11 +52,15 @@ default = [
     "transport_multilink",
     "transport_compression",
     "transport_quic",
+    "transport_serial",
     "transport_tcp",
     "transport_tls",
     "transport_udp",
     "transport_unixsock-stream",
     "transport_ws",
+    # Requires stabby, which does not build on bazel.
+    "shared-memory",
+    "unstable"
 ]
 
 [badges]
@@ -75,6 +79,7 @@ unwrap-infallible = "0.1.5"
 const_format = "0.2.32"
 zenoh = { version = "1.4.0", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "release/1.4.0", default-features = false, features = ["internal"] }
 zenoh-ext = { version = "1.4.0", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "release/1.4.0", default-features = false, features=["internal"] }
+zenoh-protocol = { version = "1.4.0", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "release/1.4.0" }
 zenoh-runtime = { version = "1.4.0", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "release/1.4.0" }
 zenoh-util = { version = "1.4.0", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "release/1.4.0" }
 flume = "*"