--- a/resource/lib.rs.em
+++ b/resource/lib.rs.em
@@ -3,8 +3,13 @@
 #![allow(clippy::upper_case_acronyms)]
 
 @[if len(msg_specs) > 0]@
+@[  if len(msg_specs) == 1]@
+#[path = "msg/@(convert_camel_case_to_lower_case_underscore(msg_specs[0][1].structure.namespaced_type.name)).rs"]
+mod msg_idiomatic;
+@[  else]@
 #[path = "msg.rs"]
 mod msg_idiomatic;
+@[  end if]@
 pub mod msg {
     pub use super::msg_idiomatic::*;
     pub mod rmw;
@@ -12,8 +17,13 @@
 @[end if]@
 
 @[if len(srv_specs) > 0]@
+@[  if len(srv_specs) == 1]@
+#[path = "srv/@(convert_camel_case_to_lower_case_underscore(srv_specs[0][1].namespaced_type.name)).rs"]
+mod srv_idiomatic;
+@[  else]@
 #[path = "srv.rs"]
 mod srv_idiomatic;
+@[  end if]@
 pub mod srv {
     pub use super::srv_idiomatic::*;
     pub mod rmw;
@@ -21,8 +31,13 @@
 @[end if]@
 
 @[if len(action_specs) > 0]@
+@[  if len(action_specs) == 1]@
+#[path = "action/@(convert_camel_case_to_lower_case_underscore(action_specs[0][1].namespaced_type.name)).rs"]
+mod action_idiomatic;
+@[  else]@
 #[path = "action.rs"]
 mod action_idiomatic;
+@[  end if]@
 pub mod action {
     pub use super::action_idiomatic::*;
     pub mod rmw;
