diff --git a/rclcpp_lifecycle/test/test_transition_wrapper.cpp b/rclcpp_lifecycle/test/test_transition_wrapper.cpp
index 39791d5d..f1b58679 100644
--- a/rclcpp_lifecycle/test/test_transition_wrapper.cpp
+++ b/rclcpp_lifecycle/test/test_transition_wrapper.cpp
@@ -119,7 +119,8 @@ TEST_F(TestTransitionWrapper, assignment_operator) {
   EXPECT_EQ(2, b->goal_state().id());
 }
 
-TEST_F(TestTransitionWrapper, exceptions) {
+// TODO(asymingt) enable once mocking works in Bazel.
+TEST_F(TestTransitionWrapper, DISABLED_exceptions) {
   rcl_lifecycle_transition_t * null_handle = nullptr;
   EXPECT_THROW((void)rclcpp_lifecycle::Transition(null_handle), std::runtime_error);
 