diff --git a/rclcpp_lifecycle/test/test_lifecycle_node.cpp b/rclcpp_lifecycle/test/test_lifecycle_node.cpp
index c31d408a..6403f808 100644
--- a/rclcpp_lifecycle/test/test_lifecycle_node.cpp
+++ b/rclcpp_lifecycle/test/test_lifecycle_node.cpp
@@ -423,7 +423,8 @@ TEST_F(TestDefaultStateMachine, trigger_transition_shutdown_label) {
   }
 }
 
-TEST_F(TestDefaultStateMachine, trigger_transition_rcl_errors) {
+// TODO(asymingt) enable once mocking works in Bazel.
+TEST_F(TestDefaultStateMachine, DISABLED_trigger_transition_rcl_errors) {
   auto test_node = std::make_shared<EmptyLifecycleNode>("testnode");
 
   {