diff --git a/rclcpp_lifecycle/test/test_lifecycle_service_client.cpp b/rclcpp_lifecycle/test/test_lifecycle_service_client.cpp
index fc3f86e5..c186a83f 100644
--- a/rclcpp_lifecycle/test/test_lifecycle_service_client.cpp
+++ b/rclcpp_lifecycle/test/test_lifecycle_service_client.cpp
@@ -448,7 +448,8 @@ protected:
   }
 };
 
-TEST_F(TestLifecycleServiceClientRCLErrors, call_services_rcl_errors) {
+// TODO(asymingt) enable once mocking works in Bazel.
+TEST_F(TestLifecycleServiceClientRCLErrors, DISABLED_call_services_rcl_errors) {
   auto lifecycle_node = std::make_shared<EmptyLifecycleNode>();
   auto lifecycle_client = std::make_shared<LifecycleServiceClient>("client_with_errors");
 