--- a/rosidl_generator_rs/__init__.py
+++ b/rosidl_generator_rs/__init__.py
@@ -17,10 +17,14 @@
 
 from pathlib import Path
 
-if os.environ['ROS_DISTRO'] <= 'humble':
+# Bazel will not have ROS_DISTRO set for build commands. Nevertheless we
+# dont' support distros before Lyrical, so this check is contextually
+# irrelevant. We have modified the line before to suppress the KeyError.
+if 'ROS_DISTRO' in os.environ and os.environ['ROS_DISTRO'] <= 'humble':
     import rosidl_cmake as rosidl_pycommon
 else:
     import rosidl_pycommon
+
 
 from rosidl_parser.definition import AbstractGenericString
 from rosidl_parser.definition import AbstractNestedType
@@ -155,55 +159,98 @@
     for action in idl_content.get_elements_of_type(Action):
         data['action_specs'].append(('action', action))
 
+    message_name = None
     if data['msg_specs']:
         for template_file, generated_filenames in mapping_msgs.items():
-            stem = _removesuffix(Path(template_file).stem, ".em")
+            if len(data['msg_specs']) == 1:
+                message_name = rosidl_pycommon.convert_camel_case_to_lower_case_underscore(
+                    data['msg_specs'][0][1].structure.namespaced_type.name)
+                stem = f"msg/{message_name}.rs"
+            else:
+                stem = _removesuffix(Path(template_file).stem, ".em")
 
             for generated_filename in generated_filenames:
-                generated_file = os.path.join(args['output_dir'],
-                                              generated_filename % stem)
+                if len(data['msg_specs']) == 1:
+                    generated_file = os.path.join(args['output_dir'],
+                                                  "msg",
+                                                  generated_filename % stem)
+                else:
+                    generated_file = os.path.join(args['output_dir'],
+                                                  generated_filename % stem)
                 rosidl_pycommon.expand_template(
-                    os.path.join(template_dir, template_file),
+                    template_file,
                     data.copy(),
                     generated_file,
                     minimum_timestamp=latest_target_timestamp)
 
+    service_name = None
     if data['srv_specs']:
         for template_file, generated_filenames in mapping_srvs.items():
-            stem = _removesuffix(Path(template_file).stem, ".em")
+            if len(data['srv_specs']) == 1:
+                service_name = rosidl_pycommon.convert_camel_case_to_lower_case_underscore(
+                    data['srv_specs'][0][1].namespaced_type.name)
+                stem = f"srv/{service_name}.rs"
+            else:
+                stem = _removesuffix(Path(template_file).stem, ".em")
 
             for generated_filename in generated_filenames:
-                generated_file = os.path.join(args['output_dir'],
-                                              generated_filename % stem)
+                if len(data['srv_specs']) == 1:
+                    generated_file = os.path.join(args['output_dir'],
+                                                  "srv",
+                                                  generated_filename % stem)
+                else:
+                    generated_file = os.path.join(args['output_dir'],
+                                                  generated_filename % stem)
                 rosidl_pycommon.expand_template(
-                    os.path.join(template_dir, template_file),
+                    template_file,
                     data.copy(),
                     generated_file,
                     minimum_timestamp=latest_target_timestamp)
 
+    action_name = None
     if data['action_specs']:
         for template_file, generated_filenames in mapping_actions.items():
-            stem = _removesuffix(Path(template_file).stem, ".em")
+            if len(data['action_specs']) == 1:
+                action_name = rosidl_pycommon.convert_camel_case_to_lower_case_underscore(
+                    data['action_specs'][0][1].namespaced_type.name)
+                stem = f"action/{action_name}.rs"
+            else:
+                stem = _removesuffix(Path(template_file).stem, ".em")
 
             for generated_filename in generated_filenames:
-                generated_file = os.path.join(args['output_dir'],
-                                              generated_filename % stem)
+                if len(data['action_specs']) == 1:
+                    generated_file = os.path.join(args['output_dir'],
+                                                  "action",
+                                                  generated_filename % stem)
+                else:
+                    generated_file = os.path.join(args['output_dir'],
+                                                  generated_filename % stem)
                 rosidl_pycommon.expand_template(
-                    os.path.join(template_dir, template_file),
+                    template_file,
                     data.copy(),
                     generated_file,
                     minimum_timestamp=latest_target_timestamp)
 
+    # If we are building for one interface, make sure to put the root library
+    # alongside the interface to support transitive merging of interfaces.
+    if message_name is not None:
+        default_lib_rs_target = os.path.join(args['output_dir'], f'msg/rust/src/msg/{message_name}.lib.rs')
+    elif service_name is not None:
+        default_lib_rs_target = os.path.join(args['output_dir'], f'srv/rust/src/srv/{service_name}.lib.rs')
+    elif action_name is not None:
+        default_lib_rs_target = os.path.join(args['output_dir'], f'action/rust/src/action/{action_name}.lib.rs')
+    else:
+        default_lib_rs_target = os.path.join(args['output_dir'], 'rust/src/lib.rs')
     rosidl_pycommon.expand_template(
         os.path.join(template_dir, 'lib.rs.em'),
         data.copy(),
-        os.path.join(args['output_dir'], 'rust/src/lib.rs'),
+        default_lib_rs_target,
         minimum_timestamp=latest_target_timestamp)
 
     cargo_toml_data = {
         'dependency_packages': dependency_packages,
         'package_name': package_name,
-        'package_version': args['package_version'],
+        'package_version': args.get('package_version', '0.0.0'),
     }
     rosidl_pycommon.expand_template(
         os.path.join(template_dir, 'Cargo.toml.em'),
