--- src/get_search_paths.cpp
+++ src/get_search_paths.cpp
@@ -15,11 +15,14 @@
 #include "ament_index_cpp/get_search_paths.hpp"
 
 #include <sys/stat.h>
+#include <unistd.h> // For readlink
+#include <limits.h> // For PATH_MAX
 
 #include <iostream>
 #include <cstdlib>
 #include <filesystem>
 #include <list>
+#include <optional>
 #include <sstream>
 #include <stdexcept>
 #include <string>
@@ -30,6 +33,36 @@
 
 namespace ament_index_cpp
 {
+
+// Cross-platform way of obtaining the full path to the current executable. Returns
+// std::nullopt if we can't determine the path. We need this to determine if a
+// .runfiles/MANIFEST file exists in the current directory.
+static std::optional<std::string>
+get_executable_path()
+{
+#if defined(_WIN32)
+  char result[MAX_PATH + 1] = {'\0'};
+  auto path_len = GetModuleFileNameA(NULL, result, MAX_PATH);
+#elif defined(__APPLE__)
+  char result[PATH_MAX + 1] = {'\0'};
+  uint32_t path_len = PATH_MAX;
+  if (_NSGetExecutablePath(result, &path_len) != 0) {
+    return std::nullopt;
+  }
+  const ssize_t resolved_path_len = readlink(result, result, PATH_MAX);
+  if (resolved_path_len > 0) {
+    result[resolved_path_len] = '\0';
+    path_len = static_cast<uint32_t>(resolved_path_len);
+  }
+#else
+  char result[PATH_MAX + 1] = {'\0'};
+  ssize_t path_len = readlink("/proc/self/exe", result, PATH_MAX);
+#endif
+  if (path_len > 0) {
+    return result;
+  }
+  return std::nullopt;
+}
 
 std::list<std::filesystem::path>
 get_searcheable_paths()
@@ -45,6 +78,25 @@
   _dupenv_s(&ament_prefix_path, &ament_prefix_path_size, env_var);
 #endif
   if (!ament_prefix_path || std::string(ament_prefix_path).empty()) {
+    // If we get here, then we have no AMENT_PREFIX_PATH, so we set the search path to the
+    // runfiles directory and rely on a symlink tree to replicate the ament index.
+    char * runfiles_dir = getenv("RUNFILES_DIR");
+    char * test_workspace = getenv("TEST_WORKSPACE");
+    if (runfiles_dir && test_workspace) {
+      const std::filesystem::path runfiles_path(runfiles_dir);
+      return std::list<std::filesystem::path>{runfiles_path / std::string(test_workspace)};
+    }
+    const std::optional<std::string> executable_path = get_executable_path();
+    if (executable_path) {
+      const std::filesystem::path runfiles_path(*executable_path + ".runfiles");
+      if (std::filesystem::exists(runfiles_path / "MANIFEST")) {
+        return std::list<std::filesystem::path>{runfiles_path / "_main"};
+      }
+    }
+    const std::filesystem::path cwd = std::filesystem::current_path();
+    if (std::filesystem::exists(cwd / ".." / "MANIFEST")) {
+      return std::list<std::filesystem::path>{cwd};
+    }
     throw std::runtime_error("Environment variable 'AMENT_PREFIX_PATH' is not set or empty");
   }
 
@@ -85,7 +137,7 @@
 {
   std::list<std::string> result;
   auto paths = get_searcheable_paths();
-  for(const auto & path : paths) {
+  for (const auto & path : paths) {
     result.push_back(path.string());
   }
   return result;
