--- a/include/pcl_conversions/pcl_conversions.h
+++ b/include/pcl_conversions/pcl_conversions.h
@@ -169,7 +169,7 @@
   void moveFromPCL(pcl::PCLImage &pcl_image, sensor_msgs::msg::Image &image)
   {
     copyPCLImageMetaData(pcl_image, image);
-    image.data.swap(pcl_image.data);
+    static_cast<std::vector<unsigned char>&>(image.data).swap(pcl_image.data);
   }
 
   inline
@@ -194,7 +194,7 @@
   void moveToPCL(sensor_msgs::msg::Image &image, pcl::PCLImage &pcl_image)
   {
     copyImageMetaData(image, pcl_image);
-    pcl_image.data.swap(image.data);
+    pcl_image.data.swap(static_cast<std::vector<unsigned char>&>(image.data));
   }
 
   /** PCLPointField <=> PointField **/
@@ -269,7 +269,7 @@
   void moveFromPCL(pcl::PCLPointCloud2 &pcl_pc2, sensor_msgs::msg::PointCloud2 &pc2)
   {
     copyPCLPointCloud2MetaData(pcl_pc2, pc2);
-    pc2.data.swap(pcl_pc2.data);
+    static_cast<std::vector<unsigned char>&>(pc2.data).swap(pcl_pc2.data);
   }
 
   inline
@@ -296,7 +296,7 @@
   void moveToPCL(sensor_msgs::msg::PointCloud2 &pc2, pcl::PCLPointCloud2 &pcl_pc2)
   {
     copyPointCloud2MetaData(pc2, pcl_pc2);
-    pcl_pc2.data.swap(pc2.data);
+    pcl_pc2.data.swap(static_cast<std::vector<unsigned char>&>(pc2.data));
   }
 
   /** pcl::PointIndices <=> pcl_msgs::PointIndices **/
