--- a/include/rosx_introspection/contrib/nanocdr.hpp
+++ b/include/rosx_introspection/contrib/nanocdr.hpp
@@ -555,11 +555,12 @@
 
 inline void Encoder::encode(const std::string& in)
 {
-  const uint32_t str_len = in.size();
+  const uint32_t str_len = in.size() + 1;
   encode(str_len);
   const auto prev_size = storage_->size();
   storage_->resize(prev_size + str_len);
-  memcpy(storage_->data() + prev_size, in.data(), str_len);
+  memcpy(storage_->data() + prev_size, in.data(), in.size());
+  (*storage_)[prev_size + in.size()] = '\0';
 }
 
 template <typename T, typename Allocator>
