--- a/include/rclcpp/typesupport_helpers.hpp
+++ b/include/rclcpp/typesupport_helpers.hpp
@@ -47,12 +47,17 @@
 /// library.
 /// \param[in] package_name The name of the package containing the type support library,
 /// e.g. "std_msgs".
+/// \param[in] interface_type The type of interface, e.g. "msg", "srv", or "action".
+/// \param[in] interface_name The camel_case name of the interface type, e.g. "compressed_image".
 /// \param[in] typesupport_identifier Type support identifier, typically "rosidl_typesupport_cpp"
 /// \throws std::runtime_error if the library is not found.
 /// \return The path to the type support library.
 RCLCPP_PUBLIC
 std::string get_typesupport_library_path(
-  const std::string & package_name, const std::string & typesupport_identifier);
+  const std::string & package_name,
+  const std::string & interface_type,
+  const std::string & interface_name,
+  const std::string & typesupport_identifier);
 
 /// \brief Load the type support library for the given type.
 /// \param[in] type The topic type, e.g. "std_msgs/msg/String"
