diff --git a/rosidl_generator_type_description/rosidl_generator_type_description/__init__.py b/rosidl_generator_type_description/rosidl_generator_type_description/__init__.py
index bbd790e..9015450 100644
--- a/rosidl_generator_type_description/rosidl_generator_type_description/__init__.py
+++ b/rosidl_generator_type_description/rosidl_generator_type_description/__init__.py
@@ -157,6 +157,7 @@ def generate_type_hash(generator_arguments_file: str) -> List[str]:
         process_include = pending_includes.pop()
         p_path = process_include.with_suffix('.json')
         pkg = p_path.parts[0]
+        print(pkg)
         pkg_dir = include_map[pkg]
         include_path = pkg_dir / p_path.relative_to(pkg)
         with include_path.open('r') as include_file: