--- a/src/rosbag2_transport/locked_priority_queue.hpp
+++ b/src/rosbag2_transport/locked_priority_queue.hpp
@@ -148,7 +148,7 @@
   std::priority_queue<typename Container::value_type, Container, StableComparator> queue_
   RCPPUTILS_TSA_GUARDED_BY(queue_mutex_);
 
-  size_t insert_sequence_number_{0} RCPPUTILS_TSA_GUARDED_BY(queue_mutex_);
+  size_t insert_sequence_number_ RCPPUTILS_TSA_GUARDED_BY(queue_mutex_){0};
 };
 
 #ifdef __APPLE__
