--- a/src/rosbag2_transport/topic_filter.cpp
+++ b/src/rosbag2_transport/topic_filter.cpp
@@ -361,8 +361,7 @@
     type_known = cache_it->second;
   } else {
     try {
-      auto package_name = std::get<0>(rclcpp::extract_type_identifier(topic_type));
-      (void)rclcpp::get_typesupport_library_path(package_name, "rosidl_typesupport_cpp");
+      (void)rclcpp::get_typesupport_library(topic_type, "rosidl_typesupport_cpp");
       type_known = true;
     } catch (std::runtime_error & e) {
       if (already_warned_unknown_types_.find(topic_type) == already_warned_unknown_types_.end()) {
