--- test/rosbag2_tests/test_rosbag2_play_end_to_end.cpp
+++ test/rosbag2_tests/test_rosbag2_play_end_to_end.cpp
@@ -45,7 +45,7 @@
       .durability(rclcpp::DurabilityPolicy::TransientLocal))
   {
     // _SRC_RESOURCES_DIR_PATH defined in CMakeLists.txt
-    bags_path_ = (fs::path(_SRC_RESOURCES_DIR_PATH) / GetParam()).generic_string();
+    bags_path_ = (fs::path("../rosbag2_tests+/resources") / GetParam()).generic_string();
     sub_ = std::make_unique<SubscriptionManager>();
     client_node_ = std::make_shared<rclcpp::Node>("test_record_client");
     cli_resume_ = client_node_->create_client<Resume>("/rosbag2_player/resume");
