diff --git a/buildrs/opaque_types_generator.rs b/buildrs/opaque_types_generator.rs
index 2dbde5f8..5145c862 100644
--- a/buildrs/opaque_types_generator.rs
+++ b/buildrs/opaque_types_generator.rs
@@ -95,7 +95,7 @@ fn produce_opaque_types_data() -> PathBuf {
         feature_args.push(feature);
     }
 
-    let _ = std::process::Command::new("cargo")
+    let _ = std::process::Command::new(std::env::var("CARGO").unwrap())
         .arg("build")
         .args(feature_args)
         .args(linker_args)