diff --git a/CMakeLists.txt b/CMakeLists.txt
index bca3da20..5fb1f023 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -258,8 +258,8 @@ file(GLOB_RECURSE rust_sources "Cargo.toml.in" "src/*.rs" "build.rs" "splitguide
 add_custom_command(
 	OUTPUT ${libs}
 	COMMAND ${CMAKE_COMMAND} -E echo \"RUSTFLAGS = $$RUSTFLAGS\"
-	COMMAND ${CMAKE_COMMAND} -E echo \"cargo ${ZENOHC_CARGO_CHANNEL} build ${cargo_flags}\"
-	COMMAND cargo ${ZENOHC_CARGO_CHANNEL} build ${cargo_flags}
+	COMMAND ${CMAKE_COMMAND} -E echo \"$ENV{CARGO} ${ZENOHC_CARGO_CHANNEL} build ${cargo_flags}\"
+	COMMAND $ENV{CARGO} ${ZENOHC_CARGO_CHANNEL} build ${cargo_flags}
 	VERBATIM
 	COMMAND_EXPAND_LISTS
 	DEPENDS "${rust_sources}"