diff --git a/tests/run_leak_check.sh b/tests/run_leak_check.sh
index 7ce533bd..22a11b45 100755
--- a/tests/run_leak_check.sh
+++ b/tests/run_leak_check.sh
@@ -2,6 +2,11 @@
 set -e
 SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
 
+# If we're in a docker container lower the ulimit to avoid a valgrind error.
+if [ -f /.dockerenv ]; then
+  ulimit -n 200000
+fi
+
 function check_leaks {
   echo "Checking $1 for memory leaks"
   valgrind --leak-check=full --num-callers=50 --log-file="$1.leaks.log" $1