--- resource/_msg.py.em
+++ resource/_msg.py.em
@@ -162,11 +162,18 @@
 @[end for]@
     }
 
+@{
+ifcode = convert_camel_case_to_lower_case_underscore(message.structure.namespaced_type.name)
+import_ifcode = locals().get('import_ifcode', ifcode)
+suffix = '__'.join(message.structure.namespaced_type.namespaces[1:]) + '__' + ifcode
+import_suffix = locals().get('import_suffix', suffix)
+}@
+
     @@classmethod
     def __import_type_support__(cls) -> None:
         try:
             from rosidl_generator_py import import_type_support  # type: ignore[attr-defined]
-            module = import_type_support('@(package_name)')
+            module = import_type_support('@(package_name)__@(import_suffix)_s__rosidl_typesupport_c')
         except ImportError:
             import logging
             import traceback
