--- rosidl_generator_py/import_type_support_impl.py
+++ rosidl_generator_py/import_type_support_impl.py
@@ -38,12 +38,11 @@
     :param pkg_name str: name of the package
     :returns: the typesupport Python module for the specified package
     """
-    module_name = '.{}_s__rosidl_typesupport_c'.format(pkg_name)
     try:
         # Since Python 3.8, on Windows we should ensure DLL directories are explicitly added
         # to the search path.
         # See https://docs.python.org/3/whatsnew/3.8.html#bpo-36085-whatsnew
         with add_dll_directories_from_env('PATH'):
-            return importlib.import_module(module_name, package=pkg_name)
+            return importlib.import_module(pkg_name)
     except ImportError:
-        raise UnsupportedTypeSupport(pkg_name)
+        raise UnsupportedTypeSupport(pkg_name)