--- ament_copyright/main.py
+++ ament_copyright/main.py
@@ -37,6 +37,16 @@
 
 
 def main(argv=sys.argv[1:]):
+    default_path = os.curdir
+    if 'TEST_SRCDIR' in os.environ and 'BAZEL_TARGET' in os.environ:
+        target = os.environ['BAZEL_TARGET']
+        repo_name = target.split('//')[0].lstrip('@')
+        if not repo_name:
+            repo_name = os.environ.get('TEST_WORKSPACE', '_main')
+        default_path = os.path.relpath(
+            os.path.join(os.environ['TEST_SRCDIR'], repo_name)
+        )
+
     extensions = [
         'c', 'cc', 'cpp', 'cxx', 'h', 'hh', 'hpp', 'hxx',
         'cmake',
@@ -49,7 +59,7 @@
     parser.add_argument(
         'paths',
         nargs='*',
-        default=[os.curdir],
+        default=[default_path],
         help='The files or directories to check. For directories files ending '
              "in %s will be considered (except directories starting with '.' "
              "or '_' and 'setup.py' files beside 'package.xml' files)." %
@@ -108,7 +118,9 @@
     if args.xunit_file:
         start_time = time.time()
 
-    filenames = get_files(args.paths, extensions, args.excludes)
+    excludes = [os.path.join(default_path, e) for e in args.excludes]
+
+    filenames = get_files(args.paths, extensions, excludes)
     if not filenames:
         print('No repository roots and files found')
 
