--- ament_mypy/main.py
+++ ament_mypy/main.py
@@ -28,6 +28,16 @@
 
 
 def main(argv: List[str] = sys.argv[1:]) -> int:
+    default_path = os.curdir
+    if 'TEST_SRCDIR' in os.environ and 'BAZEL_TARGET' in os.environ:
+        target = os.environ['BAZEL_TARGET']
+        repo_name = target.split('//')[0].lstrip('@')
+        if not repo_name:
+            repo_name = os.environ.get('TEST_WORKSPACE', '_main')
+        default_path = os.path.relpath(
+            os.path.join(os.environ['TEST_SRCDIR'], repo_name)
+        )
+
     """Command line tool for static type analysis with mypy."""
     parser = argparse.ArgumentParser(
         description='Check code using mypy',
@@ -43,7 +53,7 @@
     parser.add_argument(
         'paths',
         nargs='*',
-        default=[os.curdir],
+        default=[default_path],
         help='The files or directories to check. For directories files ending '
              "in '.py' will be considered."
     )
