--- src/shared_library.c
+++ src/shared_library.c
@@ -49,7 +49,9 @@
 #endif  // _WIN32
 
 #include "rcutils/error_handling.h"
+#include "rcutils/filesystem.h"
 #include "rcutils/macros.h"
+#include "rcutils/runfiles.h"
 #include "rcutils/shared_library.h"
 #include "rcutils/strdup.h"
 
@@ -91,11 +93,23 @@
   // for further reference.
 
 #ifndef _WIN32
-  lib->lib_pointer = dlopen(library_path, RTLD_LAZY);
+  lib->lib_pointer = dlopen(library_path, RTLD_LAZY | RTLD_GLOBAL);
+  if (NULL == lib->lib_pointer) {
+    char * runfile_name = rcutils_join_path("lib", library_path, allocator);
+    if (NULL != runfile_name) {
+      char * runfile_path = get_runfile(runfile_name, allocator);
+      allocator.deallocate(runfile_name, allocator.state);
+      if (NULL != runfile_path) {
+        lib->lib_pointer = dlopen(runfile_path, RTLD_LAZY | RTLD_GLOBAL);
+        allocator.deallocate(runfile_path, allocator.state);
+      }
+    }
+  }
   if (NULL == lib->lib_pointer) {
     RCUTILS_SET_ERROR_MSG_WITH_FORMAT_STRING("dlopen error: %s", dlerror());
     return RCUTILS_RET_ERROR;
   }
+
 
 #if defined(__APPLE__)
   const char * image_name = NULL;
@@ -152,6 +166,17 @@
 #else
   HMODULE module = LoadLibrary(library_path);
   if (!module) {
+    char * runfile_name = rcutils_join_path("lib", library_path, allocator);
+    if (NULL != runfile_name) {
+      char * runfile_path = get_runfile(runfile_name, allocator);
+      allocator.deallocate(runfile_name, allocator.state);
+      if (NULL != runfile_path) {
+        module = LoadLibrary(runfile_path);
+        allocator.deallocate(runfile_path, allocator.state);
+      }
+    }
+  }
+  if (!module) {
     RCUTILS_SET_ERROR_MSG_WITH_FORMAT_STRING(
       "LoadLibrary error: %lu", GetLastError());
     return RCUTILS_RET_ERROR;
@@ -359,4 +384,4 @@
 
 #ifdef __cplusplus
 }
-#endif
+#endif