diff --git a/rclcpp/test/rclcpp/test_node_options.cpp b/rclcpp/test/rclcpp/test_node_options.cpp
index 7abc36f3..65a40265 100644
--- a/rclcpp/test/rclcpp/test_node_options.cpp
+++ b/rclcpp/test/rclcpp/test_node_options.cpp
@@ -262,7 +262,8 @@ TEST(TestNodeOptions, append_parameter_override) {
   EXPECT_EQ(std::string("some_parameter"), options.parameter_overrides()[0].get_name());
 }
 
-TEST(TestNodeOptions, rcl_node_options_fini_error) {
+// TODO(asymingt) enable once mocking works in Bazel.
+TEST(TestNodeOptions, DISABLED_rcl_node_options_fini_error) {
   auto mock = mocking_utils::inject_on_return("lib:rclcpp", rcl_node_options_fini, RCL_RET_ERROR);
   auto options = std::make_shared<rclcpp::NodeOptions>();
   // Necessary to setup internal pointer