diff --git a/rclcpp/test/rclcpp/test_node.cpp b/rclcpp/test/rclcpp/test_node.cpp
index dc8eedb6..b58e6172 100644
--- a/rclcpp/test/rclcpp/test_node.cpp
+++ b/rclcpp/test/rclcpp/test_node.cpp
@@ -37,6 +37,7 @@
 #include "test_msgs/msg/empty.hpp"
 #include "test_msgs/srv/empty.hpp"
 
+#include "../locate_test_assets.hpp"
 #include "../mocking_utils/patch.hpp"
 
 class TestNode : public ::testing::Test
@@ -54,10 +55,10 @@ protected:
 
   void SetUp() override
   {
-    test_resources_path /= "test_node";
+    test_resources_path = get_test_asset_dir() / "test_node";
   }
 
-  std::filesystem::path test_resources_path{TEST_RESOURCES_DIRECTORY};
+  std::filesystem::path test_resources_path;
 };
 
 /*
@@ -3778,7 +3779,8 @@ TEST_F(TestNode, wait_for_graph_event) {
   EXPECT_GT(timeout, thread_completion - thread_start);
 }
 
-TEST_F(TestNode, create_sub_node_rmw_validate_namespace_error) {
+// TODO(asymingt) enable once mocking works in Bazel.
+TEST_F(TestNode, DISABLED_create_sub_node_rmw_validate_namespace_error) {
   auto node = std::make_shared<rclcpp::Node>("node", "ns");
   {
     auto mock = mocking_utils::patch_and_return(