--- test/test_flake8.py
+++ test/test_flake8.py
@@ -19,8 +19,8 @@
 @pytest.mark.flake8
 @pytest.mark.linter
 def test_flake8():
-    excluded = ['cmakelint.py']
-    rc, errors = main_with_errors(argv=['--exclude'] + excluded)
+    excluded = ['ament_lint_cmake/cmakelint.py']
+    rc, errors = main_with_errors(argv=['--exclude', ','.join(excluded)])
     assert rc == 0, \
         'Found %d code style errors / warnings:\n' % len(errors) + \
         '\n'.join(errors)
