--- a/msg/BUILD.bazel
+++ b/msg/BUILD.bazel
@@ -1,77 +1,252 @@
-# Copyright 2024 Apex.AI, Inc.
-# All rights reserved.
-
-filegroup(
-    name = "idl_messages_array",
-    srcs = [
-        "Array32.idl",
-        "Array64.idl",
-        "Array128.idl",
-        "Array256.idl",
-        "Array512.idl",
-        "Array1k.idl",
-        "Array4k.idl",
-        "Array16k.idl",
-        "Array32k.idl",
-        "Array60k.idl",
-        "Array64k.idl",
-        "Array256k.idl",
-        "Array1m.idl",
-        "Array2m.idl",
-        "Array4m.idl",
-        "Array8m.idl",
-    ],
-    visibility = ["//visibility:public"],
-)
-
-filegroup(
-    name = "idl_messages_struct",
-    srcs = [
-        "Struct16.idl",
-        "Struct256.idl",
-        "Struct4k.idl",
-        "Struct32k.idl",
-    ],
-    visibility = ["//visibility:public"],
-)
-
-filegroup(
-    name = "idl_messages_point_cloud",
-    srcs = [
-        "FixedTime.idl",
-        "FixedHeader.idl",
-        "FixedPointField.idl",
-        "PointCloud512k.idl",
-        "PointCloud1m.idl",
-        "PointCloud2m.idl",
-        "PointCloud4m.idl",
-        "PointCloud8m.idl",
-    ],
-    visibility = ["//visibility:public"],
-)
-
-filegroup(
-    name = "idl_messages_bounded_sequence",
-    srcs = [
-        "BoundedSequence1k.idl",
-        "BoundedSequence4k.idl",
-        "BoundedSequence16k.idl",
-        "BoundedSequence32k.idl",
-        "BoundedSequence64k.idl",
-        "BoundedSequence256k.idl",
-        "BoundedSequence1m.idl",
-        "BoundedSequence2m.idl",
-        "BoundedSequence4m.idl",
-        "BoundedSequence8m.idl",
-    ],
-    visibility = ["//visibility:public"],
-)
-
-filegroup(
-    name = "idl_messages_unbounded_sequence",
-    srcs = [
-        "UnboundedSequence.idl",
-        "UnboundedString.idl",
-    ],
-    visibility = ["//visibility:public"],
-)
+load("@rosidl_default_generators//:defs.bzl", "ros_idl")
+
+package(default_visibility = ["//visibility:public"])
+
+
+ros_idl(
+    name = "Array32",
+    src = "Array32.idl",
+    deps = [
+        "@//msg:",
+    ],
+)
+
+ros_idl(
+    name = "BoundedSequence1k",
+    src = "BoundedSequence1k.idl",
+)
+
+ros_idl(
+    name = "Array4k",
+    src = "Array4k.idl",
+    deps = [
+        "@//msg:",
+    ],
+)
+
+ros_idl(
+    name = "PointCloud1m",
+    src = "PointCloud1m.idl",
+)
+
+ros_idl(
+    name = "BoundedSequence256k",
+    src = "BoundedSequence256k.idl",
+)
+
+ros_idl(
+    name = "Array4m",
+    src = "Array4m.idl",
+    deps = [
+        "@//msg:",
+    ],
+)
+
+ros_idl(
+    name = "Array32k",
+    src = "Array32k.idl",
+    deps = [
+        "@//msg:",
+    ],
+)
+
+ros_idl(
+    name = "Array128",
+    src = "Array128.idl",
+    deps = [
+        "@//msg:",
+    ],
+)
+
+ros_idl(
+    name = "Array256",
+    src = "Array256.idl",
+    deps = [
+        "@//msg:",
+    ],
+)
+
+ros_idl(
+    name = "BoundedSequence1m",
+    src = "BoundedSequence1m.idl",
+)
+
+ros_idl(
+    name = "BoundedSequence32k",
+    src = "BoundedSequence32k.idl",
+)
+
+ros_idl(
+    name = "PointCloud2m",
+    src = "PointCloud2m.idl",
+)
+
+ros_idl(
+    name = "Array2m",
+    src = "Array2m.idl",
+    deps = [
+        "@//msg:",
+    ],
+)
+
+ros_idl(
+    name = "Array256k",
+    src = "Array256k.idl",
+    deps = [
+        "@//msg:",
+    ],
+)
+
+ros_idl(
+    name = "FixedTime",
+    src = "FixedTime.idl",
+)
+
+ros_idl(
+    name = "UnboundedString",
+    src = "UnboundedString.idl",
+)
+
+ros_idl(
+    name = "BoundedSequence2m",
+    src = "BoundedSequence2m.idl",
+)
+
+ros_idl(
+    name = "FixedPointField",
+    src = "FixedPointField.idl",
+)
+
+ros_idl(
+    name = "Struct32k",
+    src = "Struct32k.idl",
+)
+
+ros_idl(
+    name = "Array8m",
+    src = "Array8m.idl",
+    deps = [
+        "@//msg:",
+    ],
+)
+
+ros_idl(
+    name = "Array512",
+    src = "Array512.idl",
+    deps = [
+        "@//msg:",
+    ],
+)
+
+ros_idl(
+    name = "PointCloud8m",
+    src = "PointCloud8m.idl",
+)
+
+ros_idl(
+    name = "Struct4k",
+    src = "Struct4k.idl",
+)
+
+ros_idl(
+    name = "BoundedSequence64k",
+    src = "BoundedSequence64k.idl",
+)
+
+ros_idl(
+    name = "BoundedSequence4k",
+    src = "BoundedSequence4k.idl",
+)
+
+ros_idl(
+    name = "Array1k",
+    src = "Array1k.idl",
+    deps = [
+        "@//msg:",
+    ],
+)
+
+ros_idl(
+    name = "Array1m",
+    src = "Array1m.idl",
+    deps = [
+        "@//msg:",
+    ],
+)
+
+ros_idl(
+    name = "BoundedSequence4m",
+    src = "BoundedSequence4m.idl",
+)
+
+ros_idl(
+    name = "Struct256",
+    src = "Struct256.idl",
+)
+
+ros_idl(
+    name = "PointCloud512k",
+    src = "PointCloud512k.idl",
+)
+
+ros_idl(
+    name = "PointCloud4m",
+    src = "PointCloud4m.idl",
+)
+
+ros_idl(
+    name = "Array16k",
+    src = "Array16k.idl",
+    deps = [
+        "@//msg:",
+    ],
+)
+
+ros_idl(
+    name = "FixedHeader",
+    src = "FixedHeader.idl",
+)
+
+ros_idl(
+    name = "Array60k",
+    src = "Array60k.idl",
+    deps = [
+        "@//msg:",
+    ],
+)
+
+ros_idl(
+    name = "Struct16",
+    src = "Struct16.idl",
+)
+
+ros_idl(
+    name = "BoundedSequence8m",
+    src = "BoundedSequence8m.idl",
+)
+
+ros_idl(
+    name = "BoundedSequence16k",
+    src = "BoundedSequence16k.idl",
+)
+
+ros_idl(
+    name = "UnboundedSequence",
+    src = "UnboundedSequence.idl",
+)
+
+ros_idl(
+    name = "Array64",
+    src = "Array64.idl",
+    deps = [
+        "@//msg:",
+    ],
+)
+
+ros_idl(
+    name = "Array64k",
+    src = "Array64k.idl",
+    deps = [
+        "@//msg:",
+    ],
+)
