--- rosidl_typesupport_fastrtps_c/__init__.py
+++ rosidl_typesupport_fastrtps_c/__init__.py
@@ -17,8 +17,7 @@
 
 def generate_typesupport_fastrtps_c(generator_arguments_file):
     mapping = {
-        'idl__rosidl_typesupport_fastrtps_c.h.em':
-        'detail/%s__rosidl_typesupport_fastrtps_c.h',
-        'idl__type_support_c.cpp.em': 'detail/%s__type_support_c.cpp',
+        'idl__rosidl_typesupport_fastrtps_c.h.em': 'detail/%s__rosidl_typesupport_fastrtps_c.h',
+        'idl__type_support_c.cpp.em': 'detail/%s__rosidl_typesupport_fastrtps_c.cpp',
     }
     return generate_files(generator_arguments_file, mapping)
