--- test/rcl/test_logging.cpp
+++ test/rcl/test_logging.cpp
@@ -148,6 +148,7 @@
   }
 }
 
+// TODO(asymingt) enable once mocking works in Bazel.
 TEST(TestLogging, test_failing_external_logging_configure) {
   const char * ext_lib_flag = "--enable-" RCL_LOG_EXT_LIB_FLAG_SUFFIX;
   const char * argv[] = {"test_logging", RCL_ROS_ARGS_FLAG, ext_lib_flag};
@@ -182,6 +183,7 @@
   }
 }
 
+// TODO(asymingt) enable once mocking works in Bazel.
 TEST(TestLogging, test_failing_logger_level_configure) {
   const char * argv[] = {
     "test_logging", RCL_ROS_ARGS_FLAG,
@@ -207,6 +209,7 @@
   }
 }
 
+// TODO(asymingt) enable once mocking works in Bazel.
 TEST(TestLogging, test_failing_external_logging) {
   const char * stdout_flag = "--disable-" RCL_LOG_STDOUT_FLAG_SUFFIX;
   const char * ext_flag = "--enable-" RCL_LOG_EXT_LIB_FLAG_SUFFIX;
